### ISSUE TITLE
[v2] setNativeProps is missing

### ISSUE THREAD

**Issue by MatthieuLemoine** [styled-components/styled-components#763]
## Version
2.0.0-17

## Reproduction

```javascript
import React from 'react';
import styled from 'styled-components/native';
import { TouchableHighlight, View } from 'react-native';

const Text = styled.Text``;

export default class App extends React.Component {
  render() {
    return (
      <View>
        <TouchableHighlight>
          <Text>Test</Text>
        </TouchableHighlight>
      </View>
    );
  }
}
```

## Expected Behavior

Should not throw any errors.

## Actual Behavior

Throw the following error : 

```
invariant.js:44 Uncaught Error: Touchable child must either be native or forward setNativeProps to a native component
    at invariant (invariant.js:44)
    at ensureComponentIsNative (ensureComponentIsNative.js:20)
    at Object.componentDidMount (TouchableHighlight.js:124)
    at Object.chainedFunction [as componentDidMount] (factory.js:531)
    at ReactCompositeComponent.js:265
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at ReactCompositeComponent.js:264
    at CallbackQueue.notifyAll (CallbackQueue.js:76)
    at ReactReconcileTransaction.close (ReactReconcileTransaction.js:80)
    at ReactReconcileTransaction.closeAll (Transaction.js:206)
```

## Reason

```setNativeProps``` is missing in ```StyledNativeComponent``` in v2 branch.

**Comment by orta** [comment]
Confirmed - just hit this 👍

**Comment by philpl** [comment]
@orta It's being worked on at PR #764
